fe/fe-core/src/main/java/com/starrocks/system/PortConnectivityChecker.java

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file is based on code available under the Apache license here:
+//   https://github.com/apache/incubator-doris/blob/master/fe/fe-core/src/main/java/org/apache/doris/system/HeartbeatMgr.java
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.starrocks.system;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.starrocks.common.Config;
+import com.starrocks.common.Pair;
+import com.starrocks.common.ThreadPoolManager;
+import com.starrocks.common.util.LeaderDaemon;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.server.NodeMgr;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+/**
+ * Check the connectivity of a port, currently for edit log port and rpc_port
+ */
+public class PortConnectivityChecker extends LeaderDaemon {
+    private static final Logger LOG = LogManager.getLogger(PortConnectivityChecker.class);
+    private final List<Integer> portsToCheck = new ArrayList<>();
+    private final ExecutorService executor;
+    private Map<Pair<String, Integer>, Boolean> currentPortStates = new HashMap<>();
+
+    public PortConnectivityChecker() {
+        super("PortConnectivityChecker");
+        portsToCheck.add(Config.edit_log_port);
```

Comment:
The edit_log_port and rpc_port can be different in FE nodes. You should get the value from frontend object.